src/traits/abi.rs

```diff
@@ -0,0 +1,57 @@
+use crate::*;
+
+/// Provides a generic way of referring to and converting between a Rust object
+/// and its WinRT ABI equivalent.
+///
+/// This trait is automatically used by the generated bindings and should not be
+/// used directly.
+pub unsafe trait Abi: Sized {
+    type Abi;
```

Comment:
Can we document what constraints this associated type must fullfil? I think it must be FFI safe (aka blittable) and it must be zeroable.